3523/1/linker.rb

```diff
@@ -0,0 +1,16 @@
+require 'open-uri'
+require 'json'
+# This is Linker class. This class get links from https://rubygems.org/api/v1/gems/
+class Linker
+  attr_reader :gem_names
+  def initialize(gem_names)
+    @gem_names = gem_names
+  end
+
+  def find_links
+    @gem_names.map do |name|
+      link = "https://rubygems.org/api/v1/gems/#{name.strip}"
+      JSON.parse(URI.open(link).read)['source_code_uri']
```

Comment:
It looks very messy right now. Probably you want to make kind of pipe line, but it's no :D
I would create separate methods for hitting api and parsing json.